### PR TITLE
src/components: add component docs, cleanup props and format

### PR DESCRIPTION
**Pull request by maceksimon**
* Add component docs to all components
* Add required parameter to `card` and `cards` props in CardChallenge, CardEvent, CardListChallenge, CardListEvent, CardListProgress
* Format files with `yarn format`

### CODE REVIEW

File: src/components/CardProgress.vue [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+ * @description
+ * This component uses `useCircleSize` composable and presents progress
+ * data based on the `card` prop. It can have a default or dark variant.
+ * Border radius can be controlled by `config` parameter.
+ *
+ * Note: This component is commonly used within the `ListCardProgress`
+ * component.
+ *
+ * @props
+ * - `card`: (Object, required) The card object detailing progress.
```
Comment (tmszi) [inline src/components/CardProgress.vue:16]: cosmetic: ```- `card`: (Object, required)``` should be ```- `card` (Object, required):```

File: src/components/CardPost.vue [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+ * @description
+ * This component presents information such as post content, author, date, and
+ * other relevant details. Border radius can be controlled by `config`
+ * parameter.
+ *
+ * Note: This component is commonly used within the `ListCardPost`
+ * component.
+ *
+ * @props
+ * - `card`: (Object, required) An object containing details related to the post.
```
Comment (tmszi) [inline src/components/CardPost.vue:16]: cosmetic: ```- `card`: (Object, required)``` should be ```- `card` (Object, required):```

File: src/components/MenuLinks.vue [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+ * The `MenuLinks` component is used inside the Help Dialog window to
+ * show buttons with various outbound links.
+ *
+ * @description
+ * This component displays a list of links based on the specified variant
+ * (either social or useful links). The links are sourced from predefined
+ * mock data.
+ *
+ * @props
+ * - `title` (String): The heading or title for the list of links.
```
Comment (tmszi) [inline src/components/MenuLinks.vue:14]: Shouldn't it be a required prop?

File: src/components/MenuLinks.vue [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+ * - `variant` (String: 'social' | 'useful'): Determines the type of links
```
Comment (tmszi) [inline src/components/MenuLinks.vue:15]: Shouldn't it be a required prop or has default value?

**Review by tmszi — APPROVED** [review]

File: src/components/CardProgress.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/CardProgress.vue:16]: Fixed.

File: src/components/CardPost.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/CardPost.vue]: Fixed.

File: src/components/MenuLinks.vue (hunk 4 above)
Comment (maceksimon) [reply src/components/MenuLinks.vue]: Yes, you're right both title and variant should be required here. Fixed.

File: src/components/MenuLinks.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/MenuLinks.vue]: Yes, you're right both title and variant should be required here. Fixed.

**Pull request merged**